lib/parser/serializer.js

```diff
@@ -786,7 +786,7 @@ function serializeInto(
       const type = typeof value;
 
       // Check the key and throw error if it's illegal
-      if (!ignoreKeys.has(key)) {
+      if (!ignoreKeys.has(key) && typeof key === 'string') {
```

Comment:
prefer to check type first, before iterating through the set. This is a case where we can ensure monomorphic access to the `has` function.